source/binary.cpp

```diff
@@ -767,6 +767,13 @@ spv_result_t spvBinaryParse(const spv_const_context context, void* user_data,
                             spv_diagnostic* diagnostic) {
   spv_context_t hijack_context = *context;
   if (diagnostic) {
+    // Don't leak memory.
```

Comment:
This arguably is a bug in the caller.
Please update the method comment in libspirv.h to describe this new behaviour.

Reply:
How do you know the pointer you're trying to delete was actually created with (scalar) new?    You don't.  I think this is the wrong fix.  I think the only safe thing is to update the contract so that the caller doesn't do this.

Reply:
I reverted this fix and went about it a different way.